content/en/docs/contributing/coding-conventions.md

```diff
@@ -0,0 +1,815 @@
+}
+❌
+```
+
+#### Naming
+
+Identifiers in Unikraft use `snake case`:
+
+```c
+void my_func(void); /* Snake Case  */ ✅
+```
+
+```c
+void myFunc(void);  /* Camel Case  */ ❌
+```
+
+```c
+void MyFunc(void);  /* Pascal Case */ ❌
+```
+
+An exception to this are constants and preprocessor macros, which should use all uppercase and words separated by underscores:
+
+```c
+const int MAGIC_VALUE = 0xdeadb0b0;
+```
+
+```c
+#define IS_DEAD(bobo) ((bobo) == MAGIC_VALUE)
+```
+
+Macro names may also be regular function names if they are just wrappers for actual functions (see `uk_vprintk()` below) or if the functionality might be either implemented as macro or function (e.g., depending on architecture).
+
+### Leading Underscores
+
+A leading single underscore is **discouraged** but may be used selectively to mark identifiers which should not be used directly, like in the header `<uk/print.h>`:
+
+```c
+#if CONFIG_LIBUKDEBUG_PRINTK
+/* please use the uk_printk(), uk_vprintk() macros because they compile in the
+ * function calls only if the configured debug level requires it
+ */
+void _uk_vprintk(int lvl, const char *libname, const char *srcname,
+                 unsigned int srcline, const char *fmt, va_list ap);
+
+
+#define uk_vprintk(lvl, fmt, ap) \
+    do { \
+        if ((lvl) <= KLVL_MAX)  \
+            _uk_vprintk((lvl), __STR_LIBNAME__, __STR_BASENAME__, \
+            __LINE__, (fmt), ap); \
+    } while (0)
+```
+
+Prefer using `UK_PAGE_SIZE` instead of `_PAGE_SIZE` if required to avoid a symbol clash.
+
+However, leading underscores should **not** be used to name private functions within the same source file.
+The `static` keyword already provides the same information.
+
+```c
+static void _i_am_private(int arg)
+{
+        /* ... */
+}
+
+void i_am_public(int arg)
+{
+        _i_am_private(arg);
+}
+❌
+```
+
+Leading double underscores may only be used for:
+
+* low-level native Unikraft types like `__u64` and `__spinlock`
+* header inclusion guards
+* preprocessor macros to add another level of indirection for macro resolution
+
+{{% alert theme="info" %}}
+Rule of thumb: include `libc` headers first and then Unikraft headers.
+{{% /alert %}}
+
+* Use `uk_nonnull` instead of and `UK_NULL` instead of `__NULL`.
+
+### Prefixes
+
+Unikraft uses name prefixes to express that an identifier is part of a certain API, component or namespace.
+
+```text
+<UK API>_<COMPONENT>_<NAME>
+```
+
+Unikraft reserves the following API prefixes to describe **functions**, **function-like macros** and **data types** of the Unikraft API. This also applies (CAPITALIZED) for macros:
+
+* `uk_`  -  denotes a public Unikraft API at the library level.
+The implementation is usually found in one of the libraries in `/lib` or defined in the headers in `/include/uk`.
+For example: `uk_alloc()`
+
+* `ukplat_` - denotes a public Unikraft API, whose implementation is usually provided in `/plat` and depends on the platform (e.g., KVM) for which the Unikernel is compiled.
```

Comment:
`ukplat` and `ukarch` are subject to become obsolete by plat-rearch

Reply:
@michpappas, could you suggest a way to rephrase this? Should we drop it altogether?